src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -875,6 +876,34 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/SnapshotPreviewModel'
+    post:
+      tags:
+        - snapshots
+        - search
+        - repository
+      description: Retrieve data for a table in a snapshot.
```

Comment:
I'm just wondering why the new APIs are post methods. It seems their functions and descriptions state they retrieve.

Reply:
That's an awesome question.  There's a common misconception that POST is an operation that modifies state but it's really more of a catchall HTTP verb.   For operations that fetch those can be GET or POST but the problem with GET (well, one of the problems with GET) is that it doesn't really allow for a request body (I mean you can, but a lot of clients flat out disallow it) because it's meant to just pass in all it needs in the URL (e.g. using path or query parameters).  Some of the problems with _that_ are that
- URLs are limited in length and so you very quickly run into issues of max request length
- It's super gross to model complex queries in URL parameters and much much easier in JSON bodies

So the idea here is moving the data fetch to a POST request so that we can pass in the full query parameters.

I'll note that this is a direction that a lot of even low level web REST APIs have moved to (e.g. Elasticsearch)

Hopefully this isn't too much of a novel :). This is a great topic for our Thursday design sessions!

Reply:
@rjohanek if you're interested, we took on this work in response to a specific user pain point with the old endpoints, in case you'd like to see a concrete example of how this played out! https://broadinstitute.slack.com/archives/CD4HBRFMG/p1698249337991919